compiler/java/src/main/java/com/c/lexer/Token.java

```diff
@@ -25,7 +34,38 @@ public String getValue() {
      * @return the code position where this Token occurs (line and column number)
      */
     public String getCodePos() {
-        return "Line " + lineNum + ",Col " + colNum;
+        return "Line " + posLine + ",Col " + posCol;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Token token = (Token) o;
+        return posLine == token.posLine && posCol == token.posCol && type == token.type && value.equals(token.value);
```

Comment:
See my review comment in the lexer `expect` method. If you compare only the type there, you do not even need this method at all ...
And apart from the other discussion: I think you could have compared it by using the `hashCode` method below. That should be enough to compare two objects to equal values, right?

Reply:
I implemented `expect` like this just for the test cases and I guess that `equals()` is fine then. I will overload the `expect` method once I need it in the next layer.

Reply:
Sure 👍